gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/GenerateRestClientTask.java

```diff
@@ -156,11 +168,26 @@ public void generate()
         javaExecSpec.jvmArgs("-Dgenerator.rest.generate.version=2.0.0"); //RestRequestBuilderGenerator.run(version)
         javaExecSpec.jvmArgs("-Droot.path=" + getProject().getRootDir().getPath());
         javaExecSpec.args(_destinationDir.getAbsolutePath());
-        javaExecSpec.args(files);
+        javaExecSpec.args(sources);
       }).assertNormalExitValue()
     );
   }
 
+  /**
+   * Helper method to generate an argument file, containing one input file per line.
+   * This should be prefixed with '@' and passed to the final argument of RestRequestBuilderGenerator.
+   */
+  private File createArgFile(String prefix, List<String> files) {
```

Comment:
Although this logic is quite simple, it would probably be worthwhile to refactor this into its own utility class within `:gradle-plugins`, something like `ArgumentFileBuilder` (I say "___Builder" to stress that static utilities are unmaintainable) which handles creating the temp file and returning its "ID" (e.g. `@/something/something`). I think this would be a simple refactoring and would allow us to easily reuse it later, plus it removes complexity from this class.

Reply:
Added class `ArgumentFileGenerator`. Note that it can't live in `restli-tools` as `pegasus-plugins` has no dependency on it.